src/dashboard/get_chart_data/templates/filter_numeric.sql.jinja

```diff
@@ -0,0 +1,48 @@
+{%- import 'syntax.sql.jinja' as syntax -%}
+
+{%- set ns =namespace(matches=[]) -%}
+{%- set filters=['eq', 'ne', 'gt', 'gte', 'lt', 'lte'] -%}
+
+{%- macro render_filter( data, filter_type, bound) -%}
+{%- if filter_type == 'eq'-%}
+{{ data }} = {{ bound }}
+{%- elif filter_type == 'ne'-%}
+{{ data }} != {{ bound }}
+{%- elif filter_type == 'gt'-%}
+{{ data }} > {{ bound }}
+{%- elif filter_type == 'gte'-%}
+{{ data }} >= {{ bound }}
+{%- elif filter_type == 'lt'-%}
+{{ data }} < {{ bound }}
+{%- elif filter_type == 'lte'-%}
+{{ data }} <= {{ bound }}
+{%- else -%}
+    not found {{ filter_type }}
+{%- endif -%}
+{%- endmacro -%}
+
+{%- macro is_filter_in_list(filter_configs) -%}
+    {%- for f in filter_configs -%}
+        {%- if f.filter_type in filters -%}
+            {%- set ns.matches = ns.matches + [f] -%}
+        {%- endif -%}
+    {% endfor -%}
+{%- endmacro -%}
+
+{%- macro get_filters(filter_configs) -%}
+    {{ is_filter_in_list(filter_configs) }}
+    {%- for config in ns.matches %}
+        {{ syntax.and_delineate(loop) }}{{ render_filter(config['data'],config['filter_type'],config['bound']) }}
+    {%- endfor -%}
+    {%- set ns.matches = [] -%}
+{%- endmacro -%}
+
+{%- macro has_filter(filter_configs) %}
```

Comment:
suggestion nit: you could drop this method if you put the WHERE inside get_filters and checked the count there... :balancing hands motion:  Less flexible overall, but slightly cleaner to have the code be self-contained like that.

Reply:
this gets wonky with the multiple filter types above - my instinct was to do some sort of `string.has_filter or date.has_filter` kind of logic later on.